apps/external_ledger_db/lib/external_ledger_db/wallet.ex

```diff
@@ -0,0 +1,174 @@
+# Copyright 2017-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule ExternalLedgerDB.Wallet do
+  @moduledoc """
+  Ecto Schema representing external ledger wallets. A wallet is made up of
+  a unique address and the ID associated with it.
+  """
+  use Ecto.Schema
+  import Ecto.{Changeset, Query}
+  alias ExternalLedgerDB.Repo
+
+  @behaviour Utils.Ledgers.WalletSchema
+
+  @primary_key {:uuid, Ecto.UUID, autogenerate: true}
+  @timestamps_opts [type: :naive_datetime_usec]
+
+  @hot "hot"
+  @cold "cold"
+
+  def hot, do: @hot
+  def cold, do: @cold
+
+  @ethereum "ethereum"
+  @omg_network "omg_network"
+
+  def ethereum, do: @ethereum
+  def omg_network, do: @omg_network
+
+  schema "wallet" do
+    field(:address, :string)
+    field(:adapter, :string)
+    field(:primary, :boolean, default: false)
+    field(:type, :string)
+    field(:public_key, :string)
```

Comment:
Just some note that I’m not sure what is better, between storing keys as hex-encoded (decode on key-related operations) or store as bytes (encode on display).

Reply:
Also probably need `address` as well. Ethereum addresses for instance is _not_ public key, but the last 20 bytes of a keccak256 hash of the public key. Eventually we will need to query by address, so I think it makes sense to store address as well. (This will be returned from `Blockchain.Wallet` as `wallet_id`)

Reply:
IMHO, storing hex-encoded public key is easier for me in terms of serving external requests as I don't need to deal with conversions for every requests. Also easier for dumping, exports, etc. I don't think we'll use the public key from here as bytes anywhere anyway?

Also, `address` is the first in this field declarations list, just in case you meant something else.

Reply:
Will go with hex-encoded on a string column as this field will be used mostly for displaying & user-querying.